app/controllers/concerns/reauthentication_required_concern.rb

```diff
@@ -7,6 +9,15 @@ def confirm_recently_authenticated
     prompt_for_current_password
   end
 
+  def confirm_recently_authenticated_2fa
+    @reauthn = reauthn?
+    return unless user_fully_authenticated?
+    return if recently_authenticated? && user_session[:auth_method] != 'remember_device'
```

Comment:
general comment, no action needed, I wish we used strings less in this file, more methods like `authed_with_remember_device?`, but the file has been like this for a while